frontend/apps/web/yup-validations/jobs.ts

```diff
@@ -68,6 +68,7 @@ export const JobMappingFormValues = Yup.object({
   schema: Yup.string().required(),
   table: Yup.string().required(),
   column: Yup.string().required(),
+  id: Yup.string().optional(),
```

Comment:
It's being used in the job[id] source page to match to the incoming form mappings so we update the right element in the form mappings. The new job has a bug where, for multiple mappings, it's reverting an updated mapping if the page hasn't updated - which is what you found. But we need some identifier to match to the formMappings. Previously, you were constructing a key using the `schema.table.column` but in the new world where any of those can updated, we need a persistent, unique identifier to update. And we need that for both the new job mapping and updating an existing job mapping. 

I went the approach of an id field, but i think that actually only works for existing mappings since those formmapping fields will have an id. New job form mappings won't have an id, which i think is why that bug is happening. 

Alternatively, we can return an array of objects which will include the existing object and the new one and then use the existing object to match into the form mappings and then replace that with the new one. It would mean that we don't have to use an id or change the JobMapping interface.

Or something else?

Reply:
Let's not update the job form mapping if we can avoid it. That thing is used in so many spots I'd prefer if we didnt have to change the form itself out of fear of some regression.

You should be able to take advantage of the row index in some way shape or form to figure out which item in the form needs to be updated. Then there is no need for an identifier. I believe we are using similar logic when we do a bulk transformer update.

Reply:
updated this with the index logic @nickzelei